Clean up web endpoints

ibeckermayer opened the pull request.

Commit:
Cleans up some of the web endpoint code

zmb3 commented on lib/web/apiserver.go:
```diff
@@ -566,7 +566,7 @@ func (h *Handler) bindDefaultEndpoints() {
 	h.GET("/webapi/sites/:site/events/search/sessions", h.WithClusterAuth(h.clusterSearchSessionEvents)) // search site session events
 	h.GET("/webapi/sites/:site/sessions/:sid/events", h.WithClusterAuth(h.siteSessionEventsGet))         // get recorded session's timing information (from events)
-	h.GET("/webapi/sites/:site/sessions/:sid/stream", h.siteSessionStreamGet)                            // get recorded session's bytes (from events)
+	h.GET("/webapi/sites/:site/sessions/:sid/stream", h.WithClusterAuth(h.siteSessionStreamGet))         // get recorded session's bytes (from events)
```
Are you confident that this didn't change the behavior in any way?

ibeckermayer replied:
Good question, no I wasn't. The main difference between this endpoint and the others that get wrapped in `WithClusterAuth` is that with this endpoint, we're sending back a raw octet stream manually, whereas typically we send back a json. 

I've refactored in 27732e7b784525f641eaedab5af3e06816e92398 so that we're keeping all this authentication code centralized (the original intent of the change) while still sending back only a raw octet stream in this handler.

ibeckermayer replied:
27732e7b784525f641eaedab5af3e06816e92398

Commit:
Abstracts out authenticateRequestWithCluster so that the WithClusterAuth
logic can be used in siteSessionStreamGet without making siteSessionStreamGet
a ClusterHandler, which is expected to return a json.

Commit:
Merge branch 'master' into isaiah/cleanup-web-eps

Commit:
adds docstring

Commit:
Merge branch 'master' into isaiah/cleanup-web-eps

Commit:
Merge branch 'master' into isaiah/cleanup-web-eps

Commit:
Merge branch 'master' into isaiah/cleanup-web-eps

zmb3 committed:
Merge branch 'master' into isaiah/cleanup-web-eps
